modules/civicrm_engage/Engage/Report/Form/WalkList.php

```diff
@@ -418,10 +418,7 @@ public function executePrintmode($rows) {
 
       $dob  = $value['civicrm_contact_birth_date'];
       $age  = empty($dob) ? 0 : $this->dob2age($dob);
-      if (!empty($value['civicrm_contact_gender_id'])) {
-        $sex = $gender[CRM_Utils_Array::value('civicrm_contact_gender_id', $value)];
-      }
-      $sex  = empty($sex) ? '' : $sex;
+      $sex = $gender[$value['civicrm_contact_gender_id']] ?? '';
```

Comment:
Technically this could give a warning because the `$value['civicrm_contact_gender_id']` is evaluated first, but it's (currently) a required field so this works out.

Reply:
Maybe? This is not documented in the php docs as far as I can tell, but it was my understanding that the operator would guard against *anything* on the left side being null, but I could be wrong.

Reply:
I checked with "php -r". It does guard e.g. $x[$y][2], but not $x[$y[2]]